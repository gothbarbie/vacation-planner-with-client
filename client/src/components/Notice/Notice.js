// @flow

import * as React from 'react'
import classnames from 'classnames'

import Icon from '../Icon'

import './Notice.css'

type Props = {
  children: React.Children,
  warning?: boolean,
}

const Notice = ({ children, warning }: Props) => (
  <div
    className={classnames('notice', {
      'notice-blue': !warning,
      'notice-yellow': warning,
    })}>
    <Icon marginRight name="exclamation-circle" size="2x" />
    <div>{children}</div>
  </div>
)

export default Notice
