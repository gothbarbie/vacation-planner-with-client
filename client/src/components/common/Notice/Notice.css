.notice {
  color: #333333;
  background: #80D1FF;
  border: 2px solid #2D9CDB;
  padding: .5rem 1rem;
  font-family: 'Avenir Next', sans-serif;
  display: flex;
  align-items: center;
  margin: 2rem 0;
}

.notice .icon {
  color: #308BBF;
} 
