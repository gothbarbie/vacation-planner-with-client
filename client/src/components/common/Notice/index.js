// @flow

import React from 'react'
import type {Â Element } from 'react'

import Icon from '../Icon'

import './Notice.css'

type props = {
  children: Element<any>
}

export default ({ children } : props) => (
  <div className="notice notice-blue">
    <Icon marginRight name="exclamation-circle" size="2x" /><div>{children}</div>
  </div>
)
