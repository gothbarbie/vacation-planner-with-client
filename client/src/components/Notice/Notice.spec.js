import React from 'react'
import { shallow } from 'enzyme'

import Notice from './Notice'

describe('components/Notice', () => {
  const notice = shallow(<Notice />)

  it('renders', () => {
    expect(notice).toMatchSnapshot()
  })
})
