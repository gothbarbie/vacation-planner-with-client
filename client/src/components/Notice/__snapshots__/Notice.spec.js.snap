// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Notice renders 1`] = `
<div
  className="notice notice-blue"
>
  <Icon
    marginRight={true}
    name="exclamation-circle"
    size="2x"
  />
  <div />
</div>
`;
